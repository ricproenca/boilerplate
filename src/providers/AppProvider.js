import { ThemeProvider } from '@mui/material/styles';
import React, { Suspense } from 'react';
import { ErrorBoundary } from 'react-error-boundary';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import Store from '@Store';

import { defaultProps, propTypes } from './AppProvider.shape.js';
import ErrorBoundaryFallback from './ErrorBoundary';
import SuspenseFallback from './Suspense';
import theme from './Theme';

const AppProvider = ({ children }) => {
  // Render
  return (
    <ThemeProvider theme={theme}>
      <ErrorBoundary FallbackComponent={ErrorBoundaryFallback} onReset={() => undefined}>
        <Provider store={Store}>
          <Suspense fallback={<SuspenseFallback />}>
            <BrowserRouter>{children}</BrowserRouter>
          </Suspense>
        </Provider>
      </ErrorBoundary>
    </ThemeProvider>
  );
};

AppProvider.defaultProps = { ...defaultProps };
AppProvider.propTypes = { ...propTypes };

export default AppProvider;
