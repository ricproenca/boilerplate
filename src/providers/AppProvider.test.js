import { render } from '@testing-library/react';
import React from 'react';

import AppProvider from './AppProvider';

jest.mock('@Config/theme', () => () => ({}));

describe('AppProvider', () => {
  describe('TestingÂ rendering', () => {
    test('It matches snapshot', () => {
      // Arrange
      const props = {
        children: <div></div>
      };

      // Act
      const { asFragment } = render(<AppProvider {...props} />);

      // Assert
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
