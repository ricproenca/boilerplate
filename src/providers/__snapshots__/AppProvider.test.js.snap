// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppProvider TestingÂ rendering It matches snapshot 1`] = `
<DocumentFragment>
  <div />
   
</DocumentFragment>
`;
